app/src/main/java/org/wikipedia/history/HistoryFragment.kt

```diff
@@ -277,8 +276,16 @@ class HistoryFragment : Fragment(), BackPressedHandler {
             val voiceSearchButton = itemView.findViewById<View>(R.id.voice_search_button)
             historyFilterButton = itemView.findViewById(R.id.history_filter)
             clearHistoryButton = itemView.findViewById(R.id.history_delete)
-            searchCardView.setOnClickListener { view -> (requireParentFragment() as MainFragment).openSearchActivity(Constants.InvokeSource.NAV_MENU, null, view) }
-            voiceSearchButton.setOnClickListener { (requireParentFragment() as MainFragment).onFeedVoiceSearchRequested() }
+            searchCardView.setOnClickListener {
+                if (isAdded) {
```

Comment:
Are you sure we need all these new `isAdded` checks?  Is there any way for these listeners to run outside the lifecycle of the fragment?

Reply:
I can remove these but I think I saw some crashes happened in the SearchFragment and maybe that's a different issue.